WordPress/src/main/java/org/wordpress/android/ui/plugins/PluginDetailActivity.java

```diff
@@ -1066,4 +1113,238 @@ private boolean isPluginStateChangedSinceLastConfigurationDispatch() {
+     * will be installed after Automated Transfer is completed, but we don't need to handle anything about that.
+     *
+     * We don't know if there is any specific errors we might need to handle, so we are just showing a message about it
+     * for now.
+     *
+     * Once the transfer is initiated, we need to start checking the status of it. Check out
+     * `onAutomatedTransferStatusChecked` for the callback.
+     */
+    @SuppressWarnings("unused")
+    @Subscribe(threadMode = ThreadMode.MAIN)
+    public void onAutomatedTransferInitiated(OnAutomatedTransferInitiated event) {
+        if (isFinishing()) {
+            return;
+        }
+        if (event.isError()) {
+            handleAutomatedTransferFailed(event.error.message);
+        } else {
+            mDispatcher.dispatch(SiteActionBuilder.newCheckAutomatedTransferStatusAction(mSite));
+        }
+    }
+
+    /**
+     * After Automated Transfer is initiated, we'll need to check for the status of it several times as the process
+     * takes about 1 minute on average. We don't know if there are any specific errors we can handle, so for now we are
+     * simply showing the message.
+     *
+     * We'll get an `isCompleted` flag from the event and when that's `true` we'll need to re-fetch the site. It'll
+     * become a Jetpack site at that point and we'll need the updated site to be able to fetch the plugins and refresh
+     * this page. If the transfer is not completed, we use the current step and total steps to update the progress bar
+     * and check the status again after waiting for a second.
+     *
+     * Unfortunately we can't close the progress dialog until both the site and its plugins are fetched. Check out
+     * `onSiteChanged` for the next step.
+     */
+    @SuppressWarnings("unused")
+    @Subscribe(threadMode = ThreadMode.MAIN)
+    public void onAutomatedTransferStatusChecked(OnAutomatedTransferStatusChecked event) {
+        if (isFinishing()) {
+            return;
+        }
+        if (event.isError()) {
+            handleAutomatedTransferFailed(event.error.message);
+        } else {
+            if (event.isCompleted) {
+                // The flow is almost complete, we can show 99% complete to give us a second or so to fetch the site
+                // and its plugins
+                mAutomatedTransferProgressDialog.setProgress(99);
+                mDispatcher.dispatch(SiteActionBuilder.newFetchSiteAction(mSite));
+            } else {
+                mAutomatedTransferProgressDialog.setProgress(event.currentStep * 100 / event.totalSteps);
+                mHandler.postDelayed(new Runnable() {
+                    @Override
+                    public void run() {
+                        // Let's wait a second before checking the status again
+                        mDispatcher.dispatch(SiteActionBuilder.newCheckAutomatedTransferStatusAction(mSite));
```

Comment:
I'm wondering if we really need to recheck so soon - seems like we could recheck in a few seconds and still provide a good experience.

Reply:
We can definitely increase the duration for check, I just threw in some number there. I guess something in between 3 to 5 seconds could work?

Reply:
Fixed in e2335ba21ef1a1b1ed8b9fa4cb06abfd3c69574a.